src/compass/s1_quality.py

```diff
@@ -0,0 +1,180 @@
+                                            dataset_path_template=ds_path)
+
+            grid_group_path = '/science/SENTINEL1/CSLC/grids'
+            grid_group = h5_obj[grid_group_path]
+            x_coords_utm = grid_group['x_coordinates'][()]
+            y_coords_utm = grid_group['y_coordinates'][()]
+
+            arr = h5_obj[src_path][()]
+
+            n_in_bound = 0
+            n_invalid = 0
+            # SLOOOOOOOOWWWWWWWWWWWWWWWWW!!!!!!!!!!!!!!!!!!!!
+            for i_row, row in enumerate(arr):
+                y_coord = y_coords_utm[i_row]
+                for i_col, val in enumerate(row):
+                    coord = Point(x_coords_utm[i_col], y_coord)
+                    if boundary.contains(coord):
+                        n_in_bound += 1
+                        if isnan(val):
+                            n_invalid += 1
+
+            # write stats to HDF5
+            vals = [n_invalid, n_in_bound]
+            for ds_name, val in zip(stat_names, vals):
+                desc = f'{ds_name} of {dst_path}'
+                add_dataset_and_attrs(qa_group, Meta(ds_name, val, desc))
+
+
+def browse_image(path_h5, bursts):
+    root_path = '/science/SENTINEL1/CSLC'
+
+    # init containers source and destination paths in hdf5
+    src_paths = []
+    dst_paths = []
+
+    # add CSLCs requiring stats
+    grid_path = f'{root_path}/grids'
+
+    with h5py.File(path_h5, 'r') as h5_obj:
+        grid_group = h5_obj[grid_path]
+
+        # extract axis coords and epsg
+        x_coords_utm = grid_group['x_coordinates'][()]
+        y_coords_utm = grid_group['y_coordinates'][()]
+        epsg = grid_group['projection'][()]
+
+        # create projection to convert axis to lat lon for plot extent
+        proj = isce3.core.UTM(epsg)
+        lons = np.array([np.degrees(proj.inverse([x, y_coords_utm[0], 0])[0])
+                         for x in x_coords_utm])
+        lats = np.array([np.degrees(proj.inverse([x_coords_utm[0], y, 0])[1])
+                         for y in y_coords_utm])
+        extent = [lons[0], lons[-1], lats[0], lats[-1]]
+
+        qa_group = h5_obj[f'{root_path}/quality_assurance']
+
+        for b in bursts:
+            # get polarization to extract geocoded raster
+            pol = b.polarization
+            arr = np.abs(grid_group[pol][()])
+            arr_nan_masked = np.ma.masked_array(arr, mask=np.isnan(arr))
+
+            # prepare file output
+            date = b.sensing_start.strftime('%Y-%m-%d')
+            fname = f'{b.burst_id}_{pol}_{date}.png'
+
+            # get stats needed to set max value of plot
+            mean = np.mean(np.abs(arr_nan_masked))
+            std = np.std(np.abs(arr_nan_masked))
+            vmax = mean + 4 * std
+
+            # plot and save to disk
+            plt.close('all')
+            plt.figure(figsize=(20,10))
+            plt.imshow(np.abs(arr_nan_masked), vmax=vmax, extent=extent)
+            plt.colorbar(orientation='horizontal')
```

Comment:
If I remember correctly, browse images should not have any title and colorbar. Have you iterated with @nemo794 for the browse image specifics? Also, it might be a good idea to expose some QA options in the schema. For the browse image, I can think of 1) we want to plot the amplitude in linear or dB 2) we want to clip the image (e.g. to middle 95 percentile)